processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/DictionaryBuilding.java

```diff
@@ -53,6 +63,39 @@ public static <T> Object2IntMap<T> createReverseDictionary()
     return m;
   }
 
+  private static <T> Object2IntMap<T> createReverseDictionary(final Hash.Strategy<T> hashStrategy)
+  {
+    final Object2IntOpenCustomHashMap<T> m = new Object2IntOpenCustomHashMap<>(hashStrategy);
```

Comment:
i think this is fine for now, but i wonder if it is worth exploring/measuring using a sorted set instead like avl or rb set. if the cost difference isn't too dramatic we could instead just get by using the comparator instead of needing equals and hashcode to be implemented, which seems like it would simplify some things

Reply:
Since this PR is open for sometime, lets do this work as part of a follow up PR since we would require a some benchmark code as well.